src/gcp_scanner/arguments.py

```diff
@@ -124,6 +124,12 @@ def arg_parser():
       dest='log_file',
       help='Save logs to the path specified rather than displaying in\
  console')
+  parser.add_argument(
+      '-cc',
+      '--cpu-count',
+      default=1,
+      dest='cpu_count',
+      help='Give CPU Count limit for parallelization of crawlers.')
```

Comment:
Hmm, correct me if I am wrong here but my understanding of max_workers is that it basically asks to spin up N processes. It is up to OS to decide on how to distribute them across available CPUs. If this is correct I am not sure CPU is the right term here given that we do not control how many CPUs our scanner will actually use. It could be 10 workers using 2 CPUs or 5 workers on 3 CPUs. I'd just go with something like -n and/or --workers_count and provide users with description that it is actually number of parallel workers scanning GCP.

Reply:
I referred this [documentation](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) while writing this: 

> it will default to the number of processors on the machine.

This is the reason why I decided with the name.

> It could be 10 workers using 2 CPUs or 5 workers on 3 CPUs.

I'm also not sure regarding this. I'll have a quick look and update regarding this.

Reply:
Yes, but they do not say anything about actual CPU reserving. It is still number of workers defaulted to number of CPUs.